package/ios/React/CameraView.swift

```diff
@@ -135,10 +135,15 @@ public final class CameraView: UIView, CameraSessionDelegate, PreviewViewDelegat
         onViewReadyEvent?(nil)
       }
     } else {
+      deactivateCameraSession()
       fpsSampleCollector.stop()
     }
   }
 
+  deinit {
+    deactivateCameraSession()
+  }
```

Comment:
As I was saying in #3650, I don't like running (async) stuff in `deinit`. We have it on view unmount at the top anyways.

Can you try removing this and check if it still gracefully shuts down for your case?